src-tauri/src/mesh/device/mod.rs

```diff
@@ -139,3 +139,30 @@ impl MeshDevice {
         }
     }
 }
+
+/*
+ * Just as the MeshDevice struct contains all the information about a device (in raw packet form)
+ * the MeshGraph struct contains the network info in raw graph form. This is synchronized with
+ * the MeshDevice struct, and is used to generate the graph visualization/algorithm
+ * results (see analytics).
+ */
+
+pub struct MeshGraph {
+    pub graph: Graph,
+}
+
+impl MeshGraph {
+    pub fn new() -> Self {
+        Self {
+            graph: Graph::new(),
+        }
+    }
+}
+
+impl Clone for MeshGraph {
+    fn clone(&self) -> Self {
+        Self {
+            graph: self.graph.clone(),
```

Comment:
Why did you implement this manually? This should be able to be derived, see the `MeshDevice` struct declaration

Reply:
Our `graph_ds::Graph` (our struct, not petgraph itself) didn't implement it (and threw a ton of errors if I tried to add it implicitly). I've implemented `Clone` manually for the `Graph` instead of the `MeshDevice` struct. 

We still don't have `Serialize`, or `Deserialize`, which we'll need to send the graph to the frontend. I took a quick stab at implementing them but there's a lot of nesting (petgraph elements need to implement them too), so I'm going to put that in a different PR.

Reply:
Oh I tried to implement Serialize/Deserialize but it was hell.

Reply:
Yeah, you can start it with a `serde::Serializer` and use `serialize_field` on the elements of the `Graph` but then you have to implement `Serialize` for nodes and edges of the `petgraph` and it gets messy. Seems out of scope for this PR, but should definitely be a follow up if we're sending our graph to the frontend with the `MeshDevice` info.

Reply:
Update: serializing graph now